src/ZPublisher/WSGIPublisher.py

```diff
@@ -142,6 +142,11 @@ def transaction_pubevents(request, response, tm=transaction.manager):
         # Create new exc_info with the upgraded exception.
         exc_info = (exc_type, exc, sys.exc_info()[2])
 
+        # Raise exception from app if handle-errors is False
+        # (set by zope.testbrowser in some cases)
+        if not request.environ.get('wsgi.handleErrors', True):
+            reraise(*exc_info)
+
```

Comment:
Doesn't this have to be inside the `try/finally` block to delete the local `exc` / `exc_info` variables and avoid the reference cycle?

Reply:
I mean this kind of cycle: https://cosmicpercolator.com/2016/01/13/exception-leaks-in-python-2-and-3/